src/Components/Nav/Nav.js

```diff
@@ -0,0 +1,176 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import Category from './Components/Category';
+
+import './Nav.scss';
+
+class Nav extends Component {
+  constructor() {
+    super();
+    this.state = {
+      isCategoryShown: false,
+      navBarFixed: false,
+    };
+  }
+
+  handleMouseOnCategory = () => {
+    this.setState({
+      isCategoryShown: true,
+    });
+  };
+
+  handleMouseOffCategory = () => {
+    this.setState({
+      isCategoryShown: false,
+    });
+  };
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.scrollNavBar);
+  }
+
+  scrollNavBar = () => {
+    const scroll = window.scrollY;
+    const { navBarFixed } = this.state;
+
+    if (scroll > 150 && this.state.navBarFixed === false) {
+      this.setState({
+        navBarFixed: !navBarFixed,
+      });
+    }
+
+    if (scroll <= 150 && this.state.navBarFixed === true) {
+      this.setState({
+        navBarFixed: !navBarFixed,
+      });
+    }
+  };
```

Comment:
기능이 아직 완성되지 않으셨다고 하니, 로직만 살펴보겠습니다. 일단 조건식 두개가 있고 두 조건식은 동일한 동작을 하고 있습니다. 그렇다면 다음과 같이 정리할 수 있습니다.
```suggestion
  scrollNavBar = () => {
    const scroll = window.scrollY;
    const { navBarFixed } = this.state;
    
    const cond1 = scroll > 150 && !this.state.navBarFixed
    const cond2 = scroll <= 150 && this.state.navBarFixed
    
    if (cond1 || cond2) {
      this.setState({ navBarFixed: !navBarFixed })
      return
    }
  };
```

Reply:
와아..... 감사합니다 !!